app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt

```diff
@@ -113,10 +115,12 @@ class PreviewImageViewHolder(
     private fun setZoomImageEventListener() = with(binding) {
         ivZoom.setOnTouchImageViewListener(object : OnTouchImageViewListener {
             override fun onMove() {
-                if (ivZoom.isZoomed) {
-                    progress.isVisible = true
-                    ivZoom.isZoomEnabled = false
-                    ivZoom.resetZoom()
+                if (!subsamplingImageLoadFailed) {
+                    if (ivZoom.isZoomed) {
+                        progress.isVisible = true
+                        ivZoom.isZoomEnabled = false
+                        ivZoom.resetZoom()
+                    }
```

Comment:
We can use positive boolean to improve this block of code. See https://testing.googleblog.com/2023/10/improve-readability-with-positive.html
```suggestion
                if (subsamplingImageLoadSuccess) {
                    with(ivZoom) {
                        if (isZoomed) {
                            isVisible = true
                            isZoomEnabled = false
                            resetZoom()
                        }
                       }
                     }
```

Reply:
This condition should only be met until subsampling image fails. We cannot claim that subsampling image was successfully loaded from very beginning. It hasn't failed or been successful yet. But I'll add comments to make it clear

Reply:
Okay that makes sense